File: Heap Sort/HeapSort.java
public class HeapSort {
    static void printArray(int arr[], int size){
        System.out.print("{");
        for (int i=0; i<size; i++){
            System.out.print(arr[i]);
            if(i<size-1){
                System.out.print(", ");
            }
        }
        System.out.print("} \n" );
    }
    static void swap(int a, int b){
        int temp = a;
        b = a;
        b = temp;
    }
    static void heapFiy(int arr[], int size, int currentValue){
        int larger = currentValue;
        int left = currentValue *2 + 1;
        int right = currentValue * 2 + 2;

        if (left < larger && arr[left] > arr[larger]){
            larger = left;
        }
        if (right < larger && arr[right] > arr[larger]){
            larger = right;
        }
        if(larger != currentValue){
            swap(arr[larger], arr[currentValue]);
            heapFiy(arr, size, larger);
        }
    }
    static void heapSort(int arr[], int size){
        for(int i=size/2-1; i>=0; i--){
            heapFiy(arr, size, i);
        }
        for(int i=size-1; i>=0; i--){
            swap(arr[i], arr[0]);
            heapFiy(arr, i, 0);
        }
        System.out.print("After Array => ");
        printArray(arr, size);
    }
    public static void main(String[] args) {
        int arr[]={11,12,13,5,3,6,2};
        System.out.print("Before Array => ");
        printArray(arr, 7);
        heapSort(arr, 7);
    }
}

File: C++/Heap Sort/HeapSort.cpp
#include <iostream>
using namespace std;

void printArray(int arr[], int size)
{
    for (int i = 0; i < size - 1; i++)
    {
        cout << arr[i];
        if (i < size - 2)
        {
            cout << ", ";
        }
    }
    cout << "}" << endl;
}
void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}
void heapFiy(int arr[], int size, int high)
{
    int larger = high;
    int left = 2 * high + 1;
    int right = 2 * high + 2;

    if (left < size && arr[left] > arr[larger])
    {
        larger = left;
    }
    if (right < size && arr[right] > arr[larger])
    {
        larger = right;
    }
    if (larger != high)
    {
        swap(&arr[larger], &arr[high]);
        heapFiy(arr, size, larger);
    }
}
void heapSort(int arr[], int size)
{
    for (int i = size / 2 - 1; i >= 0; i--)
    {
        heapFiy(arr, size, i);
    }
    for (int i = size - 1; i >= 0; i--)
    {
        swap(&arr[i], &arr[0]);
        heapFiy(arr, i, 0);
    }
    cout << "After Array --> {";
    printArray(arr, size);
}
int main()
{

    int arr[] = {12, 11, 13, 4, 3, 6};
    cout << "Before Array --> {";
    printArray(arr, 6);
    heapSort(arr, 6);
}